### ISSUE TITLE
ObjectLiteralToLambda false positive when using Java interfaces with default methods

### ISSUE THREAD

**Issue by mhernand40** [detekt/detekt#4161]
When passing an anonymous object that only overrides a single method of a Java interface consisting of multiple default methods, Detekt flags an `ObjectLiteralToLambda` error. This does not occur with Kotlin interfaces consisting of multiple default functions. For example on Android, the following snippet will trigger this violation:
```Kotlin
        lifecycle.addObserver(
            object : DefaultLifecycleObserver {
                override fun onResume(owner: LifecycleOwner) {
                    // onResume logic here
                }
            }
        )
```

Where [DefaultLifecycleObserver](https://developer.android.com/reference/androidx/lifecycle/DefaultLifecycleObserver) is a Java 8 `interface` consisting of only `default` methods.

## Expected Behavior
`detektMain` should be able to distinguish between an anonymous `object` that overrides a single method of a Java interface that consists of multiple `default` methods vs. an anonymous `object` that overrides a Java functional interface (interface with only one method).

## Observed Behavior
`detektMain` does not recognize that an anonymous `object` overriding a single method of a Java interface that has multiple `default` methods cannot be converted into a lambda. It thinks that the Java interface is a functional interface consisting of only a single method.

## Steps to Reproduce
I have attached the following repro project.
[ObjectLiteralToLambdaFalsePositive.zip](https://github.com/detekt/detekt/files/7279489/ObjectLiteralToLambdaFalsePositive.zip)

Navigate into the project and run:
```
./gradlew detektMain
```

`detektMain` fails with the following error:
```
Ignoring a file detekt cannot handle: *****/ObjectLiteralToLambdaFalsePositive/lib/src/main/java/com/detekt/issues/JavaInterfaceWithDefaults.java
*****/ObjectLiteralToLambdaFalsePositive/lib/src/main/java/com/detekt/issues/MainActivity.kt:6:23: error: unresolved reference: databinding
import com.detekt.lib.databinding.ActivityMainBinding
                      ^
*****/ObjectLiteralToLambdaFalsePositive/lib/src/main/java/com/detekt/issues/MainActivity.kt:11:23: error: unresolved reference: ActivityMainBinding
        val binding = ActivityMainBinding.inflate(layoutInflater)
                      ^
*****/ObjectLiteralToLambdaFalsePositive/lib/src/main/java/com/detekt/issues/MainActivity.kt:13:26: error: variable expected
        binding.textView.isVisible = false
                         ^
style - 5min debt
        ObjectLiteralToLambda - [repro] at *****/ObjectLiteralToLambdaFalsePositive/lib/src/main/java/com/detekt/issues/Repro.kt:6:9

Overall debt: 5min


FAILURE: Build failed with an exception.
```

## Context
It is pretty common in Android development to add an anonymous [DefaultLifecycleObserver](https://developer.android.com/reference/androidx/lifecycle/DefaultLifecycleObserver) implementation that only overrides a single lifecycle event method. However, Detekt is asking the developer to convert the anonymous `object` to a lambda which is not possible since `DefaultLifecycleObserver` is not a functional interface. It has multiple default methods that can be overridden.

## Your Environment
* Version of detekt used: `1.18.1`
* Version of Gradle used (if applicable):
```
------------------------------------------------------------
Gradle 7.0.2
------------------------------------------------------------

Build time:   2021-05-14 12:02:31 UTC
Revision:     1ef1b260d39daacbf9357f9d8594a8a743e2152e

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.9 (Eclipse OpenJ9 openj9-0.23.0)
OS:           Mac OS X 10.16 x86_64
```
* Gradle scan link (add `--scan` option when running the gradle task): https://scans.gradle.com/s/l7lmwlhdmhyo6
* Operating System and version: Mac OS Big Sur version 11.6
* Link to your project (if it's a public repository): N/A
